### PR TITLE
Test with pytest-xdist

### PR DESCRIPTION
**Pull request by orf**
Re #55

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -6,6 +6,6 @@ deps =
     pip >= 6
     -rtest_requirements.txt
 commands =
-    coverage run --source pytest_asyncio -m py.test
+    coverage run --source pytest_asyncio -m py.test -n 4
```
Comment (discosultan) [inline tox.ini]: Why not use `-n auto`? It is described [here](https://github.com/pytest-dev/pytest-xdist#speed-up-test-runs-by-sending-tests-to-multiple-cpus) and imo should be the default choice unless there's a reason to fix the parallelism.

Comment (discosultan) [reply tox.ini]: Good to know about the Travis issue. Glad there's a solution 🙂

**Comment by orf** [comment]
I've updated the PR to run xdist on travis, plus made it run tests on MacOS.

**Comment by nicoddemus** [comment]
@orf unfortunately `pytest 3.7.0` is incompatible with `pytest-asyncio`: https://github.com/pytest-dev/pytest-asyncio/issues/89.

It should be fixed in `pytest 3.7.1` though (https://github.com/pytest-dev/pytest/pull/3754), so I suggest to pin to `pytest!=3.7.0` for now.

**Pull request closed without merging**